crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,141 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use log::{debug, info};
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+// TODO (#253): Make this configurable from a config file
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+}
+
+struct MPCInstance {
+    status: MPCSessionStatus,
+    /// The channel to send message to this instance
+    input_receiver: Option<mpsc::Sender<MPCInput>>,
+    pending_messages: Vec<MPCInput>,
+}
+
+impl MPCInstance {
+    fn set_active(&mut self) {
+        self.status = MPCSessionStatus::Active;
+        // TODO (#256): Replace hard coded 100 with the number of validators times 10
+        let (messages_handler_sender, messages_handler_receiver) = mpsc::channel(100);
+        self.input_receiver = Some(messages_handler_sender);
+        self.spawn_mpc_messages_handler(messages_handler_receiver);
+    }
+
+    /// Spawns an asynchronous task to handle incoming messages.
+    /// The [`MPCService`] will forward any message related to that instance to this channel.
+    fn spawn_mpc_messages_handler(&self, mut receiver: mpsc::Receiver<MPCInput>) {
+        tokio::spawn(async move {
+            while let Some(message) = receiver.recv().await {
+                // TODO (#235): Implement MPC messages handling
+            }
+        });
+    }
+
+    fn handle_message(&mut self, message: MPCInput) {
+        match self.status {
+            MPCSessionStatus::Active => {
+                if let Some(input_receiver) = &self.input_receiver {
+                    let _ = input_receiver.send(message);
+                }
+            }
+            MPCSessionStatus::Pending => {
+                self.pending_messages.push(message);
+            }
+            MPCSessionStatus::Finished => {
+                // Do nothing
+            }
+        }
+    }
+}
+
+/// Possible statuses of an MPC session:
+/// - Active: The session is currently running; new messages will be forwarded to the session.
+/// - Pending: Too many active instances are running atm; incoming messages will be queued. The session
```

Comment:
```suggestion
/// - Pending: Too many active instances are running at the moment; incoming messages will be queued. The session
```
Please don't use shortwords in documentation